atom/common/node_bindings.cc

```diff
@@ -213,19 +229,42 @@ void NodeBindings::Initialize() {
   // Explicitly register electron's builtin modules.
   RegisterBuiltinModules();
 
-  // Init node.
-  // (we assume node::Init would not modify the parameters under embedded mode).
-  // NOTE: If you change this line, please ping @codebytere or @MarshallOfSound
+  // pass non-null program name to argv so it doesn't crash
+  // trying to index into a nullptr
   int argc = 0;
   int exec_argc = 0;
-  const char** argv = nullptr;
+  const char* prog_name = "electron";
+  const char** argv = {&prog_name};
   const char** exec_argv = nullptr;
+
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  if (env->HasVar("NODE_OPTIONS")) {
+    std::vector<std::string> disallowed = {
+        "--inspect", "--inspect-brk",    "--inspect-port", "--require",
+        "-r",        "--openssl-config", "--loader",       "--use-openssl-ca"};
+
+    std::string options;
+    env->GetVar("NODE_OPTIONS", &options);
+    std::vector<std::string> parts = split_string(options);
+
+    for (std::string part : parts) {
+      bool is_allowed = std::find(disallowed.begin(), disallowed.end(), part) !=
+                        disallowed.end();
+      if (std::find(disallowed.begin(), disallowed.end(), part) !=
+          disallowed.end()) {
+        printf("This NODE_OPTION is not allowed in Electron");
```

Comment:
we need to exit gracefully here but i'm not sure the best way to do that 🤔

Reply:
IMO we should `LOG(WARNING)`, remove the option from the list and keep going. Disallowing these options is not a water-tight way to prevent these breaking options from being passed.